hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieBaseFileGroupRecordBuffer.java

```diff
@@ -147,6 +153,37 @@ public void close() {
     records.clear();
   }
 
+  /**
+   * Compares two {@link Comparable}s.  If both are numbers, converts them to {@link Long} for comparison.
+   * If one of the {@link Comparable}s is a String, assumes that both are String values for comparison.
+   *
+   * @param o1 {@link Comparable} object.
+   * @param o2 other {@link Comparable} object to compare to.
+   * @return comparison result.
+   */
+  @VisibleForTesting
+  static int compareTo(Comparable o1, Comparable o2) {
+    // TODO(HUDI-7848): fix the delete records to contain the correct ordering value type
```

Comment:
does this happen only for delete records?

Reply:
Yes, based on the test cases this only happens when the ordering field value is deserialized from the delete records.  We need to check if the existing Avro-based merging logic has done schema handling to make this work (which may also incur additional overhead).